packages/@aws-cdk/aws-servicecatalogappregistry/lib/aspects/stack-associator.ts

```diff
@@ -45,6 +47,15 @@ abstract class StackAssociatorBase implements IAspect {
     this.application.associateApplicationWithStack(node);
   }
 
+  /**
+   * Determines if the stack and application are in the same region.
+   *
+   * @param stack Cfn Stack.
+   */
+  private isSameRegion(stack: Stack): boolean {
+    return isRegionUnresolved(this.application.env.region, stack.region) || this.application.env.region === stack.region;
```

Comment:
why should an unresolved region be allowed here, instead of throwing an error? A comment block would be useful here.

Reply:
Will add a comment block explaining this scenario, and I think moving the `isRegionUnresolved()` check out to line 35's IF statement describes the scenario better.

We still want to allow environment-agnostic stacks/applications to attempt the association during deployment, because during synthesis we won't know the environment details to apply the check that the region is the same.

The `handleCrossRegionStack()` will display the necessary warnings if an environment-agnostic stack or application is provided to explain that the association might not work as expected during deployment.